src/ICSharpCode.SharpZipLib/Encryption/ZipAESStream.cs

```diff
@@ -81,8 +81,16 @@ public override int Read(byte[] buffer, int offset, int count)
 				int obtained = _stream.Read(_slideBuffer, _slideBufFreePos, lengthToRead);
 				_slideBufFreePos += obtained;
 
-				// Recalculate how much data we now have
-				byteCount = _slideBufFreePos - _slideBufStartPos;
+                if (obtained > 0 && obtained < lengthToRead)
+                {
+                    // less bytes are returned but may not be the end of file. Get more to fill the required length. Eof returns 0 bytes so this is safe
+                    int moreObtained = _stream.Read(_slideBuffer, _slideBufFreePos, lengthToRead - obtained);
```

Comment:
Why only 1 extra read? You could try reading from the inner stream until the buffer is completely filled or the inner stream returns 0 bytes. Seems to be more solid.

Reply:
It needs one extra read to fill the required length else it throws error and I don't want to change the logic of lengthToRead. This one works and I tested it. Can you help me how to fix this?

Reply:
Yes thats good. I will cancel my PR if that is working and merged. Let me know if there is a new nuget release. Thank you